test/handler/pool_test.exs

```diff
@@ -0,0 +1,150 @@
+defmodule Handler.PoolTest do
```

Comment:
It might also be nice to see a test that confirms that jobs in the pool time out if they take too long.